text/0000-extended-standard-library.md

```diff
@@ -0,0 +1,370 @@
+
+## Migrating Functionality from ESL to the Standard Library
+
+If the Rust Project decides they want to move something from the ESL into the Rust standard library, ESL developers should assist with the migration.
+
+## Naming and Namespaces
+
+Earlier it was stated that all crates in the ESL would ideally have names that:
+
+1) Clearly describe the functionality they offer
+2) Contain a reliable indicator of ESL membership
+
+For practical reasons this may not be the case from the start of the ESL, but it's something to think about working towards in the future.
+
+### Clearly Descriptive Crate Names
+
+Because namespacing for crates is not implemented, crate names, rather than source entity names, must be unique (e.g. "CreativeName" vs. "CreativeOrg::DescriptiveName"). This encourages creative naming in order to achieve uniqueness. As a result, crates coming into ESL are likely to have names that are creative rather than clearly descriptive.
+
+Clearly descriptive crate names would help Rust developers find what they need more easily. While crates can have descriptions associated with them, reading descriptions is slower than reading names.
+
+Perhaps more importantly though, it would also help with the readability of Rust code since crate names are used in Rust code frequently. It's the difference between "creative::ClientConfig" and "tls::clientConfig". This can be worked around with "use [creative] as [descriptive]" but that is more work and not always done in clear ways.
+
+### Reliably Indicating ESL Membership
+
+Lack of namespacing also make it difficult to clearly and reliably indicate membership in the ESL, where reliably means that only crates in the ESL can indicate membership in the ESL.
+
+As part of understanding and mitigating supply chain risks, Rust developers should be able to easily tell which crates come from the same source. Unfortunately, this is not easy today.
+
+Adding support for namespacing crates is the best way to improve the situation. [Rust RFC 3243](https://rust-lang.github.io/rfcs/3243-packages-as-optional-namespaces.html) has been passed but is not yet implemented.
+
+In the future, it would be good to implement RFC 3243 and get ESL crates into an ESL namespace (e.g. "esl::" or "ext_std::"). This would be an improvement over requiring GitHub organization checks, and it would allow ESL membership for dependencies to be reflected in Rust application code.
+
+## Support for no_std
+
+It should be possible, both technically and in terms of policy, for ESL crates to support no_std. This would not necessarily be feasible or make sense for all ESL crates, but it might be something to consider for some.
+
+# Appendix A
+[appendix-a]: #appendix-a
+
+## Example Policies
+
+A simplified list of example policies to use as potential starting point.
+
+### Evolving APIs / Breaking Changes
+
+ESL crates will be versioned with [Semantic Versioning](https://semver.org/).
+
+Patch and minor updates may happen at any time.
```

Comment:
Not to debate policy but just so I don't forget, there might want to be policies on this. In most cases, there is little difference between minor and update. When supporting for 2 years, there are major differences
- The more majors, the more work to backport, patch, and release
- Without majors, you can't patch. MSRV bumps and "risky" changes (major refactors) should necesisitate a minor